src/Core/AwsClient.php

```diff
@@ -4,23 +4,61 @@
 
 namespace AsyncAws\Core;
 
+use AsyncAws\Core\Credentials\CacheProvider;
+use AsyncAws\Core\Credentials\ChainProvider;
+use AsyncAws\Core\Credentials\ConfigurationProvider;
+use AsyncAws\Core\Credentials\CredentialProvider;
+use AsyncAws\Core\Credentials\IniFileProvider;
+use AsyncAws\Core\Exception\InvalidArgument;
 use AsyncAws\Core\Exception\MissingDependency;
 use AsyncAws\Core\Exception\RuntimeException;
 use AsyncAws\S3\S3Client;
 use AsyncAws\Ses\SesClient;
 use AsyncAws\Sqs\SqsClient;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
 
 /**
  * Base API client that instantiate other API classes if needed.
  *
  * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
-class AwsClient extends AbstractApi
+class AwsClient
```

Comment:
Could you elaborate here? I let it extend the AbstractApi because I didnt want to duplicate this code. 

I also want to support something like this: 

```php
use AsyncAws\Core\AwsClient;
use Symfony\Component\HttpClient\HttpClient;

$awsClient = new AwsClient(HttpClient::create(), [
    'region' => 'eu-central-1',
    'accessKeyId' => 'foo',
    'accessKeySecret' => 'bar',
]);

$result = $awsClient->request('POST', [
    'Action' => 'SendMessage',
    'MessageBody' => 'foobar'
], 
[
  /* headers */
], 
 'https://sqs.eu-central-1.amazonaws.com/5555555555/my_queue'
);
```

Reply:
Hmm.. I didn't think about that.. revert..

I reverted that change, and defined an arbitrary SignatureVersion.

In fact, the example won't work, as the request need to be signed and for that purpose needs a serviceName...

Reply:
Hm.. Yeah, I wrote the example before the code. 

We should make sure that works... somehow.. Or remove that class.